.github/workflows/release.yml

```diff
@@ -0,0 +1,61 @@
+# Upload a Python Package using Twine when a release is created
+
+name: Build
+on:
+  release:
+    types: [published]
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+permissions:
+  contents: read
+
+jobs:
+  package:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.10'
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install build twine
+    - name: Build package
+      run: python -m build
+    - name: Check package
+      run: twine check --strict dist/*
+    - name: Check env vars
+      run: |
+        echo "Triggered by: ${{ github.event_name }}"
+
+  # PyPI on release
+  pypi:
+    needs: package
+    runs-on: ubuntu-latest
+    if: github.event_name == 'release'
+    steps:
+    - name: Publish to PyPI
+      uses: pypa/gh-action-pypi-publish@release/v1
+      with:
+        user: __token__
+        password: ${{ secrets.PYPI_API_TOKEN }}
+
+  # TestPyPI on push
+  test_pypi:
+    needs: package
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push'
```

Comment:
Don't we want releases here too, so TestPyPI is always equal to, or ahead of, the live one?

Reply:
The `TestPyPI` is really only meant to be there so that when this is merged we can see if it works *at all* and then iterate until it does so that eventually the `Release` step actually works. Once it works, I'd rather just delete this altogether, `codespell` is a small project and installing the "latest and greatest" is already nearly trivial with `pip`. I think it's actually easier/more common for people to know the command for "install the latest from GitHub source" than it is "add and use this separate TestPyPI repository index" 🤷